plugin/adsk/scripts/USDMenuProc.mel

```diff
@@ -55,6 +55,117 @@ global proc mayaUsdMenu_duplicate(string $ufePath)
     }
 }
 
+global proc mayaUsdMenu_assignNewMaterial(string $ufePath, string $material)
+{
+    if (!hasPrimUpdater())
+        return;
+
+    if (size($ufePath) != 0 && size($material) != 0) {
+        string $temp = `python("import ufe;\
+                                ufePath = ufe.PathString.path('" + $ufePath + "');\
+                                item = ufe.Hierarchy.createItem(ufePath);\
+                                contextOps = ufe.ContextOps.contextOps(item);\
+                                cmd = contextOps.doOpCmd(['Assign New Material', 'USD', '" + $material + "']);\
```

Comment:
The `'USD'` part could be misleading. Looks like the context ops are only interested in items 0 and 2 in that list. Maybe you should leave that middle string blank?